arch/posix/CMakeLists.txt

```diff
@@ -39,11 +39,11 @@ else()
     target_link_options(native_simulator INTERFACE "-m64")
     target_compile_options(native_simulator INTERFACE "-m64")
   else ()
-    zephyr_compile_options(-m32)
+    zephyr_compile_options(-m32 -msse -mfpmath=sse)
```

Comment:
oh, this needs to use `-msse2` instead of `-msse` for 64-bit floats too. I'd forgotten that the original sse was 32-bit only.

Reply:
Uhm, are you sure? the issue @martinjaeger reported is solved with this switches.

Reply:
Anyhow, I used the sse2 switch

Reply:
yeah, `-msse` fixes 32-bit floats, but you need `-msse2` to also fix 64-bit floats.